test/features/router/misc.js

```diff
@@ -259,6 +259,73 @@ describe('router - misc', function() {
         assert.deepEqual(result.uri, new URI('en.wikipedia.org/path1/test1/test2/test3', {}, false));
     });
 
+    it('should support calculating a hash in template', function() {
+        var requestTemplate = new Template({
+            body: '{$.request.hash}'
+        });
+        var result = requestTemplate.eval({
+            request: {
+                method: 'post',
+                body: 'a'
+            }
+        });
+        assert.deepEqual(result.body, '575bd4981fc14132c40646e6a115e80e8fcb9618');
+    });
+
+    it('should remove x-request-id header from hash', function() {
+        var requestTemplate = new Template({
+            body: '{$.request.hash}'
+        });
+        var result1 = requestTemplate.eval({
+            request: {
+                method: 'post',
+                headers: {
+                    'x-request-id': '10'
+                },
+                body: 'a'
+            }
+        });
+        assert.deepEqual(result1.body, '19a5337cc49833b0923ac4b6d72744bf8e915de9');
```

Comment:
Instead of asserting equivalence with a specific SHA1, it'd be better to compare the hashes of `result1` and `result2`, as we are really interested in the fact they are equal.